src/chart/line/LineView.ts

```diff
@@ -161,32 +161,35 @@ function turnPointsIntoStep(
     const pt: number[] = [];
     const nextPt: number[] = [];
     for (; i < points.length - 2; i += 2) {
-        nextPt[0] = points[i + 2];
-        nextPt[1] = points[i + 3];
-        pt[0] = points[i];
-        pt[1] = points[i + 1];
-        stepPoints.push(pt[0], pt[1]);
-
-        switch (stepTurnAt) {
-            case 'end':
-                stepPt[baseIndex] = nextPt[baseIndex];
-                stepPt[1 - baseIndex] = pt[1 - baseIndex];
-                stepPoints.push(stepPt[0], stepPt[1]);
-                break;
-            case 'middle':
-                const middle = (pt[baseIndex] + nextPt[baseIndex]) / 2;
-                const stepPt2 = [];
-                stepPt[baseIndex] = stepPt2[baseIndex] = middle;
-                stepPt[1 - baseIndex] = pt[1 - baseIndex];
-                stepPt2[1 - baseIndex] = nextPt[1 - baseIndex];
-                stepPoints.push(stepPt[0], stepPt[1]);
-                stepPoints.push(stepPt2[0], stepPt2[1]);
-                break;
-            default:
-                // default is start
-                stepPt[baseIndex] = pt[baseIndex];
-                stepPt[1 - baseIndex] = nextPt[1 - baseIndex];
-                stepPoints.push(stepPt[0], stepPt[1]);
+        if (!isNaN(points[i]) && !isNaN(points[i + 1])) {      
+            pt[0] = points[i];
+            pt[1] = points[i + 1];
+            stepPoints.push(pt[0], pt[1]);
+        }
+        if (!isNaN(points[i + 2]) && !isNaN(points[i + 3])) { 
```

Comment:
Seems it may not work when there are multiple consecutive NaN values. And handling these consecutive NaN values may be too complex and easy to bring bugs.

Perhaps the original filter logic is simpler and more robust. Sorry for the wrong suggestion.

But still the variable name `points_` should be renamed to some others without underline, like `newPoints`.

Reply:
Haha, OK, I have fixed this problem. And I also dealt with filtering nulls only when using connectNull: true. 
Please review the code again，Thanks～